src/priorities-queues.js

```diff
@@ -0,0 +1,284 @@
+import Bluebird from 'bluebird'
+import { EventEmitter } from 'events'
+import CustomError from './custom-error'
+import Signal from './signal'
+
+export default class PrioritiesQueues extends EventEmitter {
+  constructor (options) {
+    super()
+
+    if (!options) {
+      throw new CustomError({
+        type: 'missing parameter',
+        message: 'This method requires an options as the first argument',
+      })
+    }
+
+    if (!options.sqs) {
+      throw new CustomError({
+        type: 'missing parameter',
+        message: 'This method requires options.sqs',
+      })
+    }
+
+    if (!options.queues || !Object.keys(options.queues).length) {
+      throw new CustomError({
+        type: 'missing parameter',
+        message: 'This method requires options.queues',
+      })
+    }
+
+    if (!options.priorities || !options.priorities.length) {
+      throw new CustomError({
+        type: 'missing parameter',
+        message: 'This method requires options.priorities',
+      })
+    }
+
+    options.priorities.forEach((queueName) => {
+      if (!options.queues[queueName]) {
+        throw new CustomError({
+          type: 'invalid parameter',
+          message: 'There are queues at priorities list that are not at the queues list',
+        })
+      }
+    })
+
+    this.options = options
+    this.customPriorities = [].concat(options.priorities)
```

Comment:
This is just to clone the priorities?

Reply:
Exactly, the current babel config won't let me destructure =\